src/rdydata.c

```diff
@@ -416,3 +416,47 @@ PetscErrorCode RDyCreatePrognosticVec(RDy rdy, Vec *prog_vec) {
   PetscCall(VecDuplicate(rdy->X, prog_vec));
   PetscFunctionReturn(PETSC_SUCCESS);
 }
+
+// reads data for a single DOF from a binary file into a global Vec
+PetscErrorCode RDyReadOneDOFGlobalVecFromBinaryFile(RDy rdy, const char filename[], Vec *global) {
```

Comment:
What about `RDyRead{Global,Local}DOFFromBinaryFile` or `RDyRead{Global,Local}ComponentFromBinaryFile`? Would we ever want to read one component from a file storing a multi-component `Vec`?

Reply:
1. By including "OneDOF" in the function name, I was trying to indicate that we need to read the data using `rdy->aux_dm` instead of `rdy->dm`. However, I'm open to a better name.
2. I don't expect we would read only one component from a file storing a multi-component `Vec`?